scripts/usb_reset.py

```diff
@@ -90,6 +89,48 @@ def get_cg_dir(domid):
     return "/sys/fs/cgroup/devices/qemu-{}".format(domid)
 
 
+def get_ids_path(device):
+    usb_dir = "/var/run/nonpersistent/usb"
+    if not os.path.isdir(usb_dir):
+        os.makedirs(usb_dir)
+    return usb_dir + "/{}".format(device)
+
+
+def save_device_ids(device):
+    path = dev_path(device)
+
+    try:
+        stat = os.stat(path)
+        ids_info = "{} {}".format(stat.st_uid, stat.st_gid)
+    except OSError as e:
+        log.error("Failed to get stat of file {}: {}".format(path, str(e)))
+        exit(1)
+
+    try:
+        with open(get_ids_path(device), "w") as f:
+            f.write(ids_info)
+    except IOError as e:
+        log.error("Failed to save of file ids {}: {}".format(path, str(e)))
+        exit(1)
+
+
+def restore_device_ids(device):
```

Comment:
I would suggest `load_device_ids` since `restore_device_ids` sounds to me like it would actually be doing the `chown` as well.

Reply:
done.